tests/Common/Utilities/CredentialRepositoryTests.cs

```diff
@@ -0,0 +1,176 @@
+            {
+                Location = Path.GetTempFileName();
+                if (!create) File.Delete(Location);
+            }
+
+            public string Location { get; }
+
+            public void Dispose()
+            {
+                try
+                {
+                    if (File.Exists(Location)) File.Delete(Location);
+                }
+                catch
+                {
+                    // Nothing to do, here
+                }
+            }
+        }
+
+        [Test]
+        [SuppressMessage("ReSharper", "ObjectCreationAsStatement")]
+        public void Construction_FileNameEmpty_ThrowsException()
+        {
+            Action ctor = () => new CredentialRepository(string.Empty);
+            ctor.Should().Throw<ArgumentException>();
+        }
+
+        [Test]
+        public void Remove_ExistingItem_ItemRemoved()
+        {
+            using var temp = new TempFile();
+            var sut = new CredentialRepository(temp.Location);
+
+            var url = "http://portal.azure.com";
+
+            // Initially add
+            var initial = new OAuthTokenCredential("[DisplayId]", "[AccessToken]", DateTime.UtcNow, "[RefreshToken]");
+            sut.Store(url, initial);
+
+            // Update
+            sut.Remove(url);
+
+            sut.TryGetCredential(url, out _).Should().BeFalse();
+        }
+
+        [Test]
+        public void Remove_NoCredentialsStoredForUrl_NothingHappen()
+        {
+            using var temp = new TempFile();
+            var sut = new CredentialRepository(temp.Location);
+
+            sut.Invoking(s => s.Remove("http://portal.azure.com")).Should().NotThrow();
+        }
+
+        [Test]
+        [SuppressMessage("ReSharper", "AssignNullToNotNullAttribute")]
+        public void Store_CredentialNull_ThrowsException()
+        {
+            using var temp = new TempFile();
+            var sut = new CredentialRepository(temp.Location);
+
+            sut.Invoking(s => s.Store("http://portal.azure.com", null)).Should().Throw<ArgumentNullException>();
+        }
+
+        [Test]
+        public void Store_HappyPath_StorageFileCreated()
+        {
+            using var temp = new TempFile(false);
+            var sut = new CredentialRepository(temp.Location);
+
+            var credential =
+                new OAuthTokenCredential("[DisplayId]", "[AccessToken]", DateTime.UtcNow, "[RefreshToken]");
+            sut.Store("http://portal.azure.com", credential);
+
+            File.Exists(temp.Location).Should().BeTrue();
+        }
+
+        [Test]
+        public void Store_UpdateExisting_ExistingItemOverwritten()
+        {
+            using var temp = new TempFile();
+            var sut = new CredentialRepository(temp.Location);
+
+            var url = "http://portal.azure.com";
+
+            // Initially add
+            var initial = new OAuthTokenCredential("[DisplayId]", "[AccessToken]", DateTime.UtcNow, "[RefreshToken]");
```

Comment:
Grosses Bienchen fuer die Testfaelle.

Fuer die Tests nutzen wir in PiWeb das Given-When-Then-Pattern (siehe hier: https://martinfowler.com/bliki/GivenWhenThen.html )

Koenntest du die Kommentare in den Testsfaellen entsprechend aendern?

Reply:
Hab ich (wobei ich solche Kommentare wie Given, When, Then) suboptimal halte, da die Tests auch so gut lesbar sind.